mocca-client/src/main/java/com/paypal/mocca/client/MoccaUtils.java

```diff
@@ -36,21 +43,60 @@ String getValue() {
     }
 
     /*
-     * Returns an optional with the Type parametrized inside the given type,
+     * Returns an optional containing the Type parameterized inside the given type,
      * if the given type is parameterized and equals to the outer reference type.
      * If it is not, an empty optional is returned.
      */
-    static Optional<Type> getInnerType(final Type type, final Class<?> outerTypeReference) {
-        if (!(type instanceof ParameterizedType)) return Optional.empty();
+    static Optional<Type> getInnerType(final Type type, final Type outerTypeReference) {
+        if (!isParameterizedType(type)) return Optional.empty();
 
         ParameterizedType parameterizedType = (ParameterizedType) type;
-        if (parameterizedType.getRawType().getTypeName().equals(outerTypeReference.getName())) {
+        if (parameterizedType.getRawType().getTypeName().equals(outerTypeReference.getTypeName())) {
             return Optional.of(parameterizedType.getActualTypeArguments()[0]);
         } else {
             return Optional.empty();
         }
     }
 
+    /*
+     * Returns the first parametrized type inside the given type.
+     * If the given type is not parameterized an IllegalArgumentException is thrown.
+     */
+    static Type getInnerType(final Type type) {
+        if (!isParameterizedType(type)) throw new IllegalArgumentException("Given type is not parameterized");
+
+        ParameterizedType parameterizedType = (ParameterizedType) type;
+        return parameterizedType.getActualTypeArguments()[0];
+    }
+
+    /*
+     * Returns true if the given type is a parameterized type.
+     */
+    static boolean isParameterizedType(Type type) {
+        if (type == null) throw new IllegalArgumentException("Type cannot be null");
+
+        return type instanceof ParameterizedType;
+    }
+
+    /*
+     * Returns true if the given type is a parameterized type
+     * and its raw type is the same as one of the given raw types
+     */
+    static boolean isParameterizedType(Type type, Type... rawTypes) {
+        if (type == null) throw new IllegalArgumentException("Type cannot be null");
+        if (rawTypes == null) throw new IllegalArgumentException("Raw type cannot be null");
+
+        if(!(type instanceof ParameterizedType)) return false;
+        Type actualRawType = ((ParameterizedType) type).getRawType();
+
+        final Set<Type> rawTypesSet = new HashSet<>();
+        for (Type t : rawTypes) rawTypesSet.add(t);
+        return rawTypesSet.contains(actualRawType);
```

Comment:
This is best-case O(n) + 0(1) + O(1).  The following is worst case O(n), best case O(1).  Am I missing something?
```suggestion
        return Arrays.asList(rawTypes).contains(actualRawType);
```
Or course, you can just use a short-circuiting for-loop as well.

Reply:
Done